### PR TITLE
[v6r13] Fixes for the GFAL2StorageBase and GFAL2_XROOTStorage plugin

### PR DESCRIPTION
**Pull request by fibbo**
## GFAL2StorageBase:

* Cleand up \__init__: removed unused member variables
* __putSingleFile checks the origin of the source file whether it's local or from and srm or root storage. 
* Moved unrelated code out of try/except (__getSingleMetadata and __listSingleDirectory)
* __getChecksum needs a checksum set by the SE (set as self.checksumType) otherwise the checksum entry in the metadata will be empty

## GFAL2_XROOTStorage
* Cleaned up \__init__: removed unused member variables
* Added a _getExtendedAttributes method until gfal2-plugin-xrootd version 0.4 is available. Added hard coded list of extended attributes because plugin returns a wrong list

### CODE REVIEW

File: Resources/Storage/GFAL2_StorageBase.py [hunk 1]
```diff
@@ -57,18 +57,12 @@ def __init__( self, storageName, parameters ):
     # # gfal2 API
     self.gfal2 = gfal2.creat_context()
     self.gfal2.set_opt_boolean( "BDII", "ENABLE", False )
-    # # save c'tor params
-    self.name = storageName
-    self.protocol = parameters['Protocol']
     self.spaceToken = parameters['SpaceToken']
-
-    # #stage limit - 12h
```
Comment (chaen) [inline Resources/Storage/GFAL2_StorageBase.py]: Why removing the comments ?

Comment (fibbo) [inline Resources/Storage/GFAL2_StorageBase.py]: I thought they didn't add much. The variables are called stageTimeout and gfal2Timeout (and one comment just said gfal2 timeout again).
I put them back in if that makes it more readable.

File: Resources/Storage/GFAL2_StorageBase.py [hunk 2]
```diff
@@ -1229,24 +1231,6 @@ def __listSingleDirectory( self, path ):
...
-            subDirs[fullPath] = metadataDict
-          elif metadataDict['File']:
-            files[fullPath] = metadataDict
-          else:
-            self.log.debug( "GFAL2_StorageBase.__listSingleDirectory: found item which is neither file nor directory", fullPath )
-
-      return S_OK( {'SubDirs' : subDirs, 'Files' : files} )
-
     except gfal2.GError, e:
       if e.code == errno.ENOENT:
```
Comment (chaen) [inline Resources/Storage/GFAL2_StorageBase.py]: What if the error is different from ENOENT ? You don't handle the case. You let it go through ?

File: Resources/Storage/GFAL2_StorageBase.py (hunk 1 above)
Comment (chaen) [inline Resources/Storage/GFAL2_StorageBase.py]: Well, put a real and correct comment then :-)

File: Resources/Storage/GFAL2_StorageBase.py (hunk 2 above)
Comment (fibbo) [inline Resources/Storage/GFAL2_StorageBase.py]: I'm not sure what you mean, for me there is also an else: clause that would catch other exceptions?

Comment (chaen) [inline Resources/Storage/GFAL2_StorageBase.py]: Ohh sorry my bad, I did not see that the diff had a gap ! Sorry !

Comment (fibbo) [inline Resources/Storage/GFAL2_StorageBase.py]: I was looking for it for a good while as well :)

**Pull request merged**